AEPServices/Sources/utility/hitprocessor/PersistentHitQueue.swift

```diff
@@ -58,8 +69,16 @@ public class PersistentHitQueue: HitQueuing {
     }
 
     /// A recursive function for processing hits, it will continue processing all the hits until none are left in the data queue
-    private func processNextHit() {
+    /// - Parameter ignoreBatchLimit: a `Bool` flag to determine batching hits
+    private func processNextHit(ignoreBatchLimit: Bool = false) {
         queue.async {
+            if self.batching && !ignoreBatchLimit {
```

Comment:
Can you add a few comments in English describing this logic? I find it a bit hard to follow just reading it.